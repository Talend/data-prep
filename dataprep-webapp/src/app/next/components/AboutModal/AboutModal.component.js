import React from 'react';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { Map } from 'immutable';
import { Dialog, Icon } from '@talend/react-components';
import { cmfConnect, Inject } from '@talend/react-cmf';
import { translate } from 'react-i18next';
import { getDefaultTranslate } from '../../../i18n';

import I18N from './../../constants/i18n';
import './AboutModal.scss';

export const DEFAULT_STATE = new Map({
	show: false,
	expanded: false,
});

export class AboutModal extends React.Component {
	static DISPLAY_NAME = 'Translate(AboutModal)';

	constructor(props) {
		super(props);
		this.close = this.close.bind(this);
		this.toggle = this.toggle.bind(this);
	}

	close() {
		this.props.setState({ show: false });
	}

	toggle() {
		this.props.setState(({ state }) => ({ expanded: !state.get('expanded', false) }));
	}

	render() {
		const { state, t } = this.props;
		const show = state.get('show', false);
		if (!show) {
			return null;
		}

		const expanded = state.get('expanded', false);
		const bar = {
			actions: {
				center: [
					{
						actionId: 'help:about:toggle',
						label: expanded ? t('tdp-app:LESS') : t('tdp-app:MORE'),
						onClick: this.toggle,
					},
				],
			},
		};

		return (
			<Inject
				component="Dialog"
				header={t('tdp-app:ABOUT_MODAL_TITLE')}
				type={Dialog.TYPES.INFORMATIVE}
				onHide={this.close}
				actionbar={bar}
				show={show}
			>
				<Icon name="talend-tdp-colored" className={'about-logo'} />
				<div className="about-excerpt">
					<div>{t('tdp-app:ABOUT_VERSION_NAME', { version: this.props.displayVersion })}</div>
					<div>{t('tdp-app:COPYRIGHT', { year: new Date().getFullYear() })}</div>
				</div>
				{expanded && (
					<table className={'about-versions'}>
						<thead>
							<tr>
								<th>{t('tdp-app:SERVICE')}</th>
								<th>{t('tdp-app:BUILD_ID')}</th>
								<th>{t('tdp-app:VERSION')}</th>
							</tr>
						</thead>
						<tbody>
							{this.props.services.map((service) => {
								const srv = service.toJS();
								return (
									<tr>
										<td>{srv.serviceName}</td>
										<td>{srv.buildId}</td>
										<td>{srv.versionId}</td>
									</tr>
								);
							})}
						</tbody>
					</table>
				)}
			</Inject>
		);
	}
}

AboutModal.displayName = 'AboutModal';
AboutModal.propTypes = {
	state: ImmutablePropTypes.contains({ show: PropTypes.bool }).isRequired,
	setState: PropTypes.func.isRequired,
	...cmfConnect.propTypes,
};
AboutModal.defaultProps = {
	t: getDefaultTranslate,
};

export default translate(I18N.TDP_APP_NAMESPACE)(AboutModal);
