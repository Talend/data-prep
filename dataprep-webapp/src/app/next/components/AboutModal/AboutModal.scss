.modal-body {
	text-align: center;
	padding: {
		top: 10px;
		left: 32px;
		right: 32px;
	}

	.about-logo {
		width: 5em;
		height: 5em;
	}

	.about-excerpt {
		margin-top: 20px;
	}

	.services-versions {
		margin: 0 auto;
		margin-top: 30px;
		text-align: justify;

		th {
			text-transform: uppercase;
		}

		td+td, th+th {
			border-left: 80px solid transparent;
		}
	}
}
