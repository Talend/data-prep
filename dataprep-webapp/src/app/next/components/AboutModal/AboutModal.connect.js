import { cmfConnect } from '@talend/react-cmf';
import { Map } from 'immutable';
import AboutModal from './AboutModal.component';

export default cmfConnect({
	defaultState: new Map({ show: false }),
})(AboutModal);
