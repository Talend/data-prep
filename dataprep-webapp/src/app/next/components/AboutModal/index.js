import AboutModal from './AboutModal.connect';

export default AboutModal;
