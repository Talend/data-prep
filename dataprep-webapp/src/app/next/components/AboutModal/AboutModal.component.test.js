import React from 'react';
import Immutable from 'immutable';
import { shallow } from 'enzyme';
import { AboutModal } from './AboutModal.component';

const SERVICES = Immutable.fromJS(
	['API', 'Dataset', 'Preparation', 'Transformation'].map(serviceName => ({
		versionId: '2.8.0-SNAPSHOT',
		buildId: '87d0dcd-12e0d6f',
		serviceName,
	}))
);

describe('Container(AboutModal)', () => {
	it('should render', () => {
		const state = new Immutable.Map({ show: true, expanded: false });
		const wrapper = shallow(
			<AboutModal services={SERVICES} state={state} displayVersion="mock version" />,
		);
		expect(wrapper.getElement()).toMatchSnapshot();
	});

	it('should render in expanded mode', () => {
		const state = new Immutable.Map({ show: true, expanded: true });
		const wrapper = shallow(
			<AboutModal services={SERVICES} state={state} displayVersion="mock version" />,
		);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
