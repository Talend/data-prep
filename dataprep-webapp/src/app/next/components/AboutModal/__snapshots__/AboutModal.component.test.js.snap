// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container(AboutModal) should render 1`] = `
<Inject
  actionbar={
    Object {
      "actions": Object {
        "center": Array [
          Object {
            "actionId": "help:about:toggle",
            "label": "tdp-app:MORE",
            "onClick": [Function],
          },
        ],
      },
    }
  }
  component="Dialog"
  header="tdp-app:ABOUT_MODAL_TITLE"
  onHide={[Function]}
  show={true}
  type="informative"
>
  <Icon
    className="about-logo"
    name="talend-tdp-colored"
  />
  <div
    className="about-excerpt"
  >
    <div>
      tdp-app:ABOUT_VERSION_NAME
    </div>
    <div>
      tdp-app:COPYRIGHT
    </div>
  </div>
</Inject>
`;

exports[`Container(AboutModal) should render in expanded mode 1`] = `
<Inject
  actionbar={
    Object {
      "actions": Object {
        "center": Array [
          Object {
            "actionId": "help:about:toggle",
            "label": "tdp-app:LESS",
            "onClick": [Function],
          },
        ],
      },
    }
  }
  component="Dialog"
  header="tdp-app:ABOUT_MODAL_TITLE"
  onHide={[Function]}
  show={true}
  type="informative"
>
  <Icon
    className="about-logo"
    name="talend-tdp-colored"
  />
  <div
    className="about-excerpt"
  >
    <div>
      tdp-app:ABOUT_VERSION_NAME
    </div>
    <div>
      tdp-app:COPYRIGHT
    </div>
  </div>
  <table
    className="about-versions"
  >
    <thead>
      <tr>
        <th>
          tdp-app:SERVICE
        </th>
        <th>
          tdp-app:BUILD_ID
        </th>
        <th>
          tdp-app:VERSION
        </th>
      </tr>
    </thead>
    <tbody>
      Immutable.List [
        <tr>
          <td>
            API
          </td>
          <td>
            87d0dcd-12e0d6f
          </td>
          <td>
            2.8.0-SNAPSHOT
          </td>
        </tr>,
        <tr>
          <td>
            Dataset
          </td>
          <td>
            87d0dcd-12e0d6f
          </td>
          <td>
            2.8.0-SNAPSHOT
          </td>
        </tr>,
        <tr>
          <td>
            Preparation
          </td>
          <td>
            87d0dcd-12e0d6f
          </td>
          <td>
            2.8.0-SNAPSHOT
          </td>
        </tr>,
        <tr>
          <td>
            Transformation
          </td>
          <td>
            87d0dcd-12e0d6f
          </td>
          <td>
            2.8.0-SNAPSHOT
          </td>
        </tr>,
      ]
    </tbody>
  </table>
</Inject>
`;
